mkdocs/nav.py

```diff
@@ -14,18 +14,35 @@
 log = logging.getLogger(__name__)
 
 
-def filename_to_title(filename):
+def file_to_title(filename):
     """
     Automatically generate a default title, given a filename.
     """
     if utils.is_homepage(filename):
         return 'Home'
 
-    title = os.path.splitext(filename)[0]
-    title = title.replace('-', ' ').replace('_', ' ')
-    # Captialize if the filename was all lowercase, otherwise leave it as-is.
-    if title.lower() == title:
-        title = title.capitalize()
+    table_of_contents = None
+    try:
+        with open(filename, 'r') as f:
+            lines = f.readlines()
+            html_content, table_of_contents, meta = utils.convert_markdown('\n'.join(lines), ['toc'])
```

Comment:
If you add the following line here:

    return table_of_contents.items[0].title

Then you can simplify this a bit and remove the `table_of_contents = None` and the if statement below.